examples/classification/demo_knn.py

```diff
@@ -0,0 +1,132 @@
+    """
+    assert len(new_y) == len(verification_y)
+    length = len(new_y)
+    count = 0
+    for index in range(length):
+        if new_y[index] == verification_y[index].item():
+            count += 1
+    return count / length
+
+
+def create_fold(dataset_x, dataset_y, size, seed=None):
+    """
+    Randomly splits the dataset into two parts for cross-validation.
+    Parameters
+    ----------
+    dataset_x : ht.DNDarray
+        data vectors, required
+    dataset_y : ht.DNDarray
+        labels for dataset_x, required
+    size : int
+        the size of the split to create
+    seed: int, optional
+        seed for the random generator, allows deterministic testing
+    Returns
+    ----------
+    fold_x : ht.DNDarray
+        DNDarray of shape (size,) containing data vectors from dataset_x
+    fold_y : ht.DNDarray
+        DNDarray of shape(size,) containing labels from dataset_y
+    verification_x : ht.DNDarray
+        DNDarray of shape(len(dataset_x - size),) containing all items from dataset_x not in fold_x
+    verification_y : ht.DNDarray
+        DNDarray of shape(len(dataset_y - size),) containing all items from dataset_y not in fold_y
+    """
+    assert len(dataset_y) == len(dataset_x)
+    assert size < len(dataset_x)
+
+    data_length = len(dataset_x)
+    if seed:
+        ht.random.seed(seed)
+    indices = ht.random.randint(low=0, high=data_length - 1, size=(size,))
+    indices = ht.unique(indices, sorted=True)
+    while len(indices) < size:
+        diff = size - len(indices)
+        additional = ht.random.randint(low=0, high=data_length, size=(diff,))
+        indices = ht.concatenate((indices, additional))
+        indices = ht.unique(indices, sorted=True)
+    indices = indices.tolist()
+    all_indices = [index for index in range(data_length)]
+    verification_indices = [index for index in all_indices if index not in indices]
+
+    fold_x = ht.array(dataset_x[indices], is_split=0)
+    fold_y = ht.array(dataset_y[indices], is_split=0)
+    verification_y = ht.array(dataset_y[verification_indices], is_split=0)
+    verification_x = ht.array(dataset_x[verification_indices], is_split=0)
+    return fold_x, fold_y, verification_x, verification_y
+
+
+def verify_algorithm(x, y, split_number, split_size, k, seed=None):
+    """
+    Parameters
+    ----------
+    x : ht.DNDarray
+        array containing data vectors
+    y : ht.DNDarray
+        array containing the labels for x (must be in same order)
+    split_number: int
+        the number of test iterations
+    split_size : int
+        the number of vectors used by the KNN-Algorithm
+    k : int
+        The number of neighbours for KNN-Algorithm
+    seed : int
+        Seed for the random generator used in creating folds. Used for deterministic testing purposes.
+    Returns
+    -------
+    accuracies : ht.DNDarray
+        array of shape (split_number,) containing the accuracy per run
+    """
+    assert len(x) == len(y)
```

Comment:
Please avoid assertions. They will result in AssertionError that is indistinguishable from the actual problem. Please make conditional checks and raise Value/Type/...Errors instead